File: src/main/kotlin/cz/uhk/ppro/mhjp/resftulgallery/dto/UserDto.kt
package cz.uhk.ppro.mhjp.resftulgallery.dto

class NewUserDto(
        val username: String = "",
        val password: String = "",
        val repeatPassword: String = ""
)

class UpdateUserDto(
        val username: String = "",
        val name: String = "",
        val private: Boolean = false
)

class UpdateUserPasswordDto(
        val oldPassword: String = "",
        val newPassword: String = "",
        val repeatNewPassword: String = ""
)

open class UserDataDto(
        val username: String,
        val name: String,
        val dateJoined: Long,
        val enabled: Boolean,
        val private: Boolean,
        val images: List<DataDto>
) : DataDto()

open class ListUserDataDto(
        val username: String
) : DataDto()

File: src/main/bobril/cz/uhk/ppro/mhjp/gallery/api/dto/UserDto.ts
import {DataDto} from "./baseDto";

export interface NewUserDto {
        username: string ,
        password: string ,
        repeatPassword: string
}

export interface UpdateUserDto{
        username: string ,
        name: string ,
        private: boolean
}

export interface UpdateUserPasswordDto{
        oldPassword: string,
        newPassword: string,
        repeatNewPassword: string
}

export interface UserDataDto extends DataDto{
        username: string,
        name: string,
        dateJoined: number,
        enabled: boolean,
        private: boolean,
        images: DataDto[]
}

export interface ListUserDataDto extends DataDto{
        username: string
}